Sources/Sentry/SentrySessionTracker.m

```diff
@@ -215,4 +249,16 @@ - (void)willTerminate
     self.wasDidBecomeActiveCalled = NO;
 }
 
+- (BOOL)isAppActive
+{
+    BOOL isAppActive = NO;
+#if SENTRY_HAS_UIKIT
+    SentryUIApplication *application = SentryDependencyContainer.sharedInstance.application;
+    if (application.applicationState == UIApplicationStateActive) {
+        isAppActive = YES;
+    }
+#endif
```

Comment:
`m`: We could make this work on macOS by using the NSApplication with something like
```
NSApplication *application = [NSApplication sharedApplication];
if (application.isActive) {
    isAppActive = YES;
}
```
But maybe it makes sense to also create a wrapper class as `SentryUIApplication` for testability. It's also acceptable to leave out this edge case for now and follow up.

Reply:
Instead of a wrapper I created a protocol `SentryApplication.h` which is implemented by `SentryUIApplication.h` and `SentryNSApplication.h`, so that the dependency container does not have to care about multiple types. What do you think about this approach?